src/app/interceptors/api-rewrite-url.interceptor.ts

```diff
@@ -0,0 +1,43 @@
+import type { HttpEvent, HttpHandler, HttpInterceptor, HttpRequest } from '@angular/common/http';
+import { Inject, Injectable } from '@angular/core';
+import type { Observable } from 'rxjs';
+
+import type { ApiConfiguration } from '../tokens';
+import { APIS_TOKENS, AvailableApis } from '../tokens';
+import { NotInAvailableApisError } from './errors';
+
+@Injectable()
+export class ApiRewriteUrlInterceptor<TResult> implements HttpInterceptor {
+  public constructor(@Inject(APIS_TOKENS) private readonly apisTokens: Record<AvailableApis, ApiConfiguration>) {}
+
+  private doesNotNeedRewrite(requestedUrl: string): boolean {
+    return !requestedUrl.startsWith('@');
+  }
+
+  private fullUrlFromConfiguration(apiConfiguration: ApiConfiguration, pathInfo: string = ''): string {
+    const prefix: string = apiConfiguration.prefix == null ? '' : `/${apiConfiguration.prefix}`;
+
+    return `${apiConfiguration.domain}${prefix}/${pathInfo}`;
+  }
+
+  private guardIsAvailableApi(apiUrlKey: string = ''): AvailableApis {
+    const apiKey: AvailableApis = apiUrlKey as unknown as AvailableApis;
+    if (!Object.values(AvailableApis).includes(apiKey)) throw new NotInAvailableApisError(apiUrlKey);
+
+    return apiKey;
+  }
+
+  private rewriteUrl(requestedUrl: string): string {
+    const captureUrlGroups: RegExp = /(?<apiUrlKey>.*?)\/(?<pathInfo>.+)/u;
+    const capturedGroups: RegExpExecArray | null = captureUrlGroups.exec(requestedUrl);
+    const apiKey: AvailableApis = this.guardIsAvailableApi(capturedGroups?.groups?.['apiUrlKey']);
+
+    return this.fullUrlFromConfiguration(this.apisTokens[apiKey], capturedGroups?.groups?.['pathInfo']);
+  }
+
+  public intercept(request: HttpRequest<TResult>, next: HttpHandler): Observable<HttpEvent<TResult>> {
+    if (this.doesNotNeedRewrite(request.url)) return next.handle(request);
+
+    return next.handle(request.clone({ url: this.rewriteUrl(request.url) }));
```

Comment:
Avoid negative conditionals => https://gist.github.com/wojteklu/73c6914cc446146b8b533c0988cf8d29

```suggestion
    if (this.needRewrite(request.url)) return next.handle(request.clone({ url: this.rewriteUrl(request.url) }));

    return next.handle(request);
```
```suggestion
    return this.needRewrite(request.url)
      ? next.handle(request.clone({ url: this.rewriteUrl(request.url) }))
      : next.handle(request);
```